specification/devopsinfrastructure/Microsoft.DevOpsInfrastructure.Management/main.tsp

```diff
@@ -748,3 +774,30 @@ model ImageVersionProperties {
 interface ImageVersions {
   listByImage is ArmResourceListByParent<ImageVersion>;
 }
+
+@doc("The reason code explaining why the name is unavailable. Will be null if the name is available.")
+union CheckNameAvailabilityReason {
+  string,
+
+  @doc("The name is invalid.")
+  "Invalid",
+
+  @doc("The name already exists.")
+  "AlreadyExists",
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "No ProvisioningState"
+@doc("The CheckNameAvailability operation response.")
+model CheckNameAvailabilityResult {
+  @doc("True if the name is available, otherwise false.")
+  available: boolean;
+
```

Comment:
consider using enums instead : 

[https://github.com/Azure/azure-rest-api-specs/pull/31422/checks?check_run_id=32572988597](https://github.com/Azure/azure-rest-api-specs/pull/31422/checks?check_run_id=32572988597)

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md)